platform/ios/src/MGLMapView.mm

```diff
@@ -1114,7 +1124,29 @@ - (void)updateFromDisplayLink:(CADisplayLink *)displayLink
         [self updateAnnotationViews];
         [self updateCalloutView];
         
-        [self.glView display];
+#ifdef MGL_RECREATE_GL_IN_AN_EMERGENCY
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 120200)
+        // See https://github.com/mapbox/mapbox-gl-native/issues/14232
+        // glClear can be blocked for 1 second. This code is an "escape hatch",
+        // an attempt to detect this situation and rebuild the GL views.
+        if (self.enablePresentsWithTransaction)
+        {
+            CFTimeInterval before = CACurrentMediaTime();
+            [self.glView display];
+            CFTimeInterval after = CACurrentMediaTime();
+            
+            if (after-before >= 1.0) {
```

Comment:
My second comment on this #ifdef is that I'm concerned about unintended consequences and believe the risks could outweigh the potential benefits, especially since this issue is so difficult to reproduce, but glClear could be blocked due to a variety of circumstances and we probably don't want to throw this hammer at everything. But I'll hold judgment until I understand the conditions that this #ifdef can be triggered. If this is only triggered if a developer voluntarily opts-in to use the "escape hatch", that could be ok.

Reply:
The developer would need to be building from source & provide the `MGL_RECREATE_GL_IN_AN_EMERGENCY` define as @1ec5 mentions above. 

However, we will need to test that code path regardless.

Reply:
A single OpenGL call should take milliseconds. If `glClear()` is blocked for a second or more, that’s a sign of trouble, whether or not it’s the specific issue we’re trying to work around.

Reply:
> But I remain skeptical that this is going to be a net-better experience for most apps and their end-users, compared to the status quo where sometimes when you rapidly switch between tabs many times, the map freezes and you have to restart the app.

Totally agree. We could decide (later) to change from an `#ifdef` to a developer set property, but hopefully a better solution will have shown its head by then.

Reply:
Yep, longer term it might be worth having some thing more robust anyway. But then, Metal.

Reply:
Sounds good!